clientv3/integration/black_hole_test.go

```diff
@@ -72,6 +72,9 @@ func TestBalancerUnderBlackholeKeepAliveWatch(t *testing.T) {
 	// endpoint can switch to eps[1] when it detects the failure of eps[0]
 	cli.SetEndpoints(eps...)
 
+	// give enough time for balancer resolution
+	time.Sleep(5 * time.Second)
```

Comment:
anyway we can make this more deterministic? we should avoid random wait.

Reply:
@gyuho just for curiosity, what prompted you to add this wait - any random test failure? or more like we should have it because keepalive can need some extra time (similar to what's mentioned here https://github.com/MakMukhi/grpc-go/blob/d9541d7591265dc9d2f07848e2a198adf1ba66e4/transport/transport_test.go#L467) Thanks!

Reply:
It's not for keepalive. New gRPC somehow takes a bit longer to resolve endpoints when we call `SetEndpoints`. And it makes this test flaky. I think we can go ahead and keep investigating the flaky tests for the next month. /cc @jpbetz

Reply:
@gyuho thanks!!